src/Microsoft.DocAsCode.MarkdigEngine/Validation/ValidationExtension.cs

```diff
@@ -25,6 +28,13 @@ public void Setup(MarkdownPipelineBuilder pipeline)
 
             pipeline.DocumentProcessed += document =>
             {
+                if (InclusionContext.IsInclude 
+                && (string.Equals(Path.GetExtension(InclusionContext.RootFile?.ToString()), ".yml", StringComparison.OrdinalIgnoreCase) 
+                || string.Equals(Path.GetExtension(InclusionContext.RootFile?.ToString()), ".yaml", StringComparison.OrdinalIgnoreCase)))
+                {
+                    document.SetData("SchemaName", YamlMime.ReadMime(InclusionContext.RootFile?.ToString()));
```

Comment:
`null.ToString()` will throw exception. Skip `SetData` when RootFile is null.

Reply:
InclusionContext.RootFile should not be null, or it would drop into this line. But you are right, that ReadMime could be null, so let me address this.